crates/bvs-vault-bank-tokenized/tests/integration_test.rs

```diff
@@ -0,0 +1,1733 @@
+
+        for staker in stakers.iter() {
+            let query_shares = QueryMsg::Shares {
+                staker: staker.to_string(),
+            };
+            let shares: Uint128 = vault.query(&mut app, &query_shares).unwrap();
+
+            // shares stays the same
+            assert_eq!(shares, Uint128::new(original_deposit_amount));
+
+            let query_assets = QueryMsg::Assets {
+                staker: staker.to_string(),
+            };
+            let assets: Uint128 = vault.query(&mut app, &query_assets).unwrap();
+
+            // assets should be halved
+            assert_eq!(assets, Uint128::from(50_000_000_u128));
+        }
+    }
+
+    //negative test
+    {
+        // non-callable address
+        let msg = ExecuteMsg::SlashLocked(Amount(Uint128::new(original_deposit_amount)));
+        let resp = vault.execute(&mut app, &stakers[0], &msg);
+        assert!(resp.is_err());
+
+        // larger than vault balance
+        let msg = ExecuteMsg::SlashLocked(Amount(Uint128::new(original_deposit_amount * 3)));
+        let resp = vault.execute(&mut app, router.addr(), &msg);
+        assert!(resp.is_err());
+
+        // zero amount
+        let msg = ExecuteMsg::SlashLocked(Amount(Uint128::new(0)));
+        let resp = vault.execute(&mut app, router.addr(), &msg);
+        assert!(resp.is_err());
+    }
+}
+
+#[test]
+fn test_deposit_transfer_then_withdraw_to() {
+    let (mut app, tc) = TestContracts::init();
+    let vault = tc.vault;
+
+    let staker = app.api().addr_make("staker/1");
+    let beneficiary = app.api().addr_make("beneficiary/1");
+    let initial_deposit_amount: u128 = 30_000_000;
+    let donation_amount = 1000;
+    let owner = app.api().addr_make("owner");
+
+    app.send_tokens(owner.clone(), staker.clone(), &coins(100_000_000, "denom"))
+        .unwrap();
+
+    let msg = ExecuteMsg::DepositFor(RecipientAmount {
+        recipient: staker.clone(),
+        amount: Uint128::new(initial_deposit_amount),
+    });
+    vault
+        .execute_with_funds(
+            &mut app,
+            &staker,
+            &msg,
+            coins(initial_deposit_amount, "denom"),
+        )
+        .unwrap();
+
+    {
+        let staker_balance = app.wrap().query_balance(&staker, "denom").unwrap();
+        assert_eq!(staker_balance, coin(70_000_000, "denom")); // unstaked_capital
+
+        let contract_balance = app.wrap().query_balance(vault.addr(), "denom").unwrap();
+        assert_eq!(contract_balance, coin(30_000_000, "denom")); // staked_capital
+
+        let query_shares = QueryMsg::Shares {
+            staker: staker.to_string(),
+        };
+
+        let shares: Uint128 = vault.query(&mut app, &query_shares).unwrap();
+        assert_eq!(shares, Uint128::new(30_000_000)); // initial_deposit_amount
+    }
+
+    // Transfer to beneficiary
+    {
+        let msg = ExecuteMsg::Transfer {
+            recipient: beneficiary.to_string(),
+            amount: Uint128::new(donation_amount),
```

Comment:
```suggestion
            amount: Uint128::new(donation_amount),
```

Why are you using `donation_amount` here? 
Semantically, it's confusing.

Reply:
Cause original staker is donating his receipt token to random address. Feel free to suggest though.